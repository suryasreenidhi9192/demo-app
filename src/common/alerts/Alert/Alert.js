import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import {
    SvgNotificationSuccessLight,
    SvgNotificationInfoLight,
    SvgNotificationWarningLight,
    SvgNotificationErrorLight,
    SvgCloseX
} from '../../icons'

import './Alert.scss'

/**
 * @public
 * Builds a dismissable Alert that displays important messages by type
 * @param {React|String}  params.footer        React Component or string to render at the bottom of the Alert
 * @param {String}        params.title         Title placed above message in Alert body
 * @param {String}        params.type          Error info etc. Used to toggle style of Alert
 * @param {Boolean}       params.retain        Retain, do not allow dismiss of Alert
 * @param {Boolean}       params.scroll        Alert should scroll window to the top of the page. Needed for long forms.
 * @param {Function}      params.handleClose   By default handleClose toggles internal state.
 *                                             But it can be overridden when it appears inside a <Modal>
*/
class Alert extends PureComponent {

    constructor(props) {
        super(props)
        this.state = {
            alertDismissed: false,
            scrolled: false
        }
    }

    componentDidMount() {
        this.conditionallyScroll()
    }

    componentDidUpdate() {
        this.conditionallyScroll()
    }

    conditionallyScroll() {
        const { scroll = false } = this.props
        const { scrolled } = this.state

        if (scroll && !scrolled) {
            window.scrollTo(0, 0)
            this.setState({ scrolled: true })
        }
    }

    getIcon(type) {

        switch (type) {

            case 'success':
                return <SvgNotificationSuccessLight className="alert-icon" />

            case 'error':
                return <SvgNotificationErrorLight className="alert-icon" />

            case 'warning':
                return <SvgNotificationWarningLight className="alert-icon" />

            case 'info':
                return <SvgNotificationInfoLight className="alert-icon" />

            default:
                return <SvgNotificationErrorLight className="alert-icon" />

        }

    }

    render() {

        const {
            children = '',
            type = 'error',
            title = '',
            footer = '',
            retain = false,
            handleClose = () => { this.setState({ alertDismissed: true }) }
        } = this.props

        const isDismissed = (this.state.alertDismissed) ? 'hidden' : ''
        const icon = this.getIcon(type)

        return (

            <div className={ `alert ${type} ${isDismissed}` } role="alert">

                <div className={`alert-header ${type}`}>
                    <span className="header" />
                </div>

                <div className="alert-body">
                    <ul className="layout">
                        <li>
                            { icon }
                        </li>
                        <li>
                            { !!title && <div className="title">{ title }</div> }
                            <div className="message">
                                { children }
                            </div>
                        </li>
                        <li onClick={ handleClose } >
                            { !retain && <SvgCloseX className="close-icon" /> }
                        </li>
                    </ul>

                    <div className="alert-footer">{ footer }</div>

                </div>

            </div>
        )
    }
}

Alert.propTypes = {
    /** React Component or string to render in the body */
    children: PropTypes.node,
    /** React Component or string to render at the bottom of the Alert */
    footer: PropTypes.node,
    /** Title placed above message in Alert body */
    title: PropTypes.string,
    /** Error info etc. Used to toggle style of Alert */
    type: PropTypes.string,
    /** Retain, do not allow dismiss of Alert */
    retain: PropTypes.bool,
    /** Alert should scroll window to the top of the page. Needed for long forms. */
    scroll: PropTypes.bool,
    /** By default handleClose toggles internal state. But it can be overridden when it appears inside a <Modal> */
    handleClose: PropTypes.func
};

export default Alert
