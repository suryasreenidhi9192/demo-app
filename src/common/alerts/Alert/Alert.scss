@import "../../shared/css/variables";

.alert {

    width: auto;
    margin: 16px 0;
    border: var(--border);
    border-radius: 2px;
    overflow: hidden;
    box-shadow: var(--box-shadow-near);

    .alert-header {
        height: 6px;
        &.success { background-color: var(--utility-green); }
        &.error   { background-color: var(--utility-red); }
        &.warning { background-color: var(--utility-amber); }
        &.info    { background-color: var(--utility-blue); }
    }

    .alert-body {

        background-color: var(--content-background);

        .layout {
            align-items: flex-start;
            display: flex;
            padding: 16px;
            line-height: normal;
            > li:nth-child(1) { flex: 0 0 24px; }
            > li:nth-child(2) { flex: 1 1 auto; margin: 0 16px; }
            > li:nth-child(3) { flex: 0 0 24px; }
        }

        .alert-icon {
            width: 24px; height: 24px;
        }

        .title {
            font-family: $lt;
            font-size: 20px;
            line-height: 24px;
            margin-bottom: 8px;
        }

        .message {
            font-family: $lt;
            font-size: 16px;
            line-height: 24px;
        }

        .close-icon {
            display: inline-block;
            padding: 4px;
            width: 16px;
            height: 16px;
            fill: var(--action-alt);
            cursor: pointer;
            transition: fill 0.3s;

            &:hover {
                fill: var(--action-alt-hover);
            }

            &:active {
                fill: var(--action-alt-active);
                transition: fill 0.05s;
            }
        }

        .alert-footer {
            display: flex; justify-content: center;
            background-color: var(--content-footer-background);
            border-top: var(--border);
            padding: 16px;
            &:empty { display: none; }
        }

    }

}

