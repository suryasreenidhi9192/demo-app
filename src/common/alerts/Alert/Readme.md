### Simple Alert
```jsx
    <Alert>Hello Alert</Alert>
```

### Info Alert with Much Copy
```jsx
    <Alert type="info">Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy foster collaborative thinking to further the overall value proposition. Organically grow the holistic world view of disruptive innovation via workplace diversity and empowerment.</Alert>

```

### Warning Alert with Footer Emoji
```jsx
    <Alert
        type="warning"
        footer="Footer copy 😀 😎 👍 💯"
    >
        Onward and upward, productize the deliverables and focus on the bottom line time to open the kimono loop back, so organic growth, yet we need to crystallize a plan for feature creep shoot me an email.
    </Alert>
```

### Success Alert with Header Emoji
```jsx
    <Alert
        type="success"
    >
        Some text...
    </Alert>
```

### Error Alert with Footer Button
```jsx
    const Button = require('../../buttons/Button').default;
    <Alert
        type="error"
        footer={
                <Button display="secondary">
                    Click Me!
                </Button>
            }
    >
        Not enough bandwidth I just wanted to give you a heads-up cannibalize, or turd polishing. Criticality we don't want to boil the ocean.
    </Alert>
```
