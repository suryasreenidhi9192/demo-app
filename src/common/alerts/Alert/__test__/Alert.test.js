import React from 'react'
import { mount } from 'enzyme';
import { matchSnapshot } from '../../../../../__test__/__utils__'

import Alert from '../'

const props = {
    retain: false,
    scroll: false,
    children: (<b>This is some copy for the body of the Alert</b>)
}

const errorProps = {
    ...props,
    type: 'error'
}

const successProps = {
    ...props,
    type: 'success'
}

const warningProps = {
    ...props,
    type: 'warning'
}

const infoProps = {
    ...props,
    type: 'info'
}

const headerProps = {
    ...props,
    header: (<h1>Heading</h1>)
}

const footerProps = {
    ...props,
    footer: (<h6>Footer</h6>)
}

const testDismiss = () => {
    const alert = mount(
        <Alert { ...props } />
    )
    alert.find('.layout li:last-child').simulate('click')
    setTimeout(() => {
        expect(alert.find('.hidden').exists()).toBe(true)
    }, 0)
}

const testRetain = () => {
    const retainProps = {
        ...props,
        retain: true
    }
    expect(mount(<Alert { ...retainProps } />).find('.close-icon').exists())
        .toBe(false)
}

const testScroll = () => {
    const scrollProps = {
        ...props,
        scroll: true
    }
    window.scrollTo = jest.fn()
    mount(<Alert { ...scrollProps } />)
    expect(window.scrollTo)
        .toHaveBeenCalled()
}

describe('Alert:.......................SnapShot Testing:', () => {
    test(
        'Error Alert Matches jest snapshot',
        matchSnapshot(Alert, errorProps)
    )
    test(
        'Success Alert Matches jest snapshot',
        matchSnapshot(Alert, successProps)
    )
    test(
        'Warning Alert Matches jest snapshot',
        matchSnapshot(Alert, warningProps)
    )
    test(
        'Info Alert Matches jest snapshot',
        matchSnapshot(Alert, infoProps)
    )
    test(
        'Alert with footer matches jest snapshot',
        matchSnapshot(Alert, footerProps)
    )
    test(
        'Alert with header matches jest snapshot',
        matchSnapshot(Alert, headerProps)
    )
})

describe('Alert:.......................Prop Rendering Logic:', () => {
    test(
        'Alert with retain prop has no dismiss button',
        testRetain
    )
    test(
        'Alert with scroll will scroll to the top',
        testScroll
    )
})

describe('Alert:.......................Event Handling:', () => {
    test(
        'Clicking the dismiss button closes the Alert',
        testDismiss
    )
})
