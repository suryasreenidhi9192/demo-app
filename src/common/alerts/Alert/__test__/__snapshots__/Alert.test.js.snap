// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert:.......................SnapShot Testing: Alert with footer matches jest snapshot 1`] = `
<div
  className="alert error "
  role="alert"
>
  <div
    className="alert-header error"
  >
    <span
      className="header"
    />
  </div>
  <div
    className="alert-body"
  >
    <ul
      className="layout"
    >
      <li>
        <svg
          className="alert-icon"
          viewBox="0 0 32 32"
        >
          <title>
            
        Error Notification
    
          </title>
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              className="Notification-Error-Light_svg__path-01"
              d="M21.744 29.867l8.123-8.123V10.256l-8.123-8.123H10.256l-8.123 8.123v11.488l8.123 8.123h11.488zM8.956 31.583l-8.331-8.33c-.4-.4-.625-.944-.625-1.51V10.257c0-.565.225-1.108.625-1.508L8.748.625c.4-.4.943-.625 1.508-.625h11.488c.566 0 1.109.225 1.509.625l8.122 8.123c.4.4.625.943.625 1.508v11.488c0 .565-.225 1.108-.625 1.508l-8.122 8.123c-.4.4-.943.625-1.509.625H9.962c-.377 0-.74-.15-1.006-.417z"
            />
            <path
              className="Notification-Error-Light_svg__path-02"
              d="M17.27 20.64a.825.825 0 01-.145.488.52.52 0 01-.171.16.356.356 0 01-.174.045h-1.56c-.133 0-.248-.068-.345-.205a.825.825 0 01-.145-.489l-.508-12.834c0-.188.049-.351.145-.488.097-.137.212-.206.345-.206h2.576c.133 0 .248.069.345.206.096.137.145.3.145.488L17.27 20.64zM16 25.6a1.422 1.422 0 110-2.844 1.422 1.422 0 010 2.844z"
            />
          </g>
        </svg>
      </li>
      <li>
        <div
          className="message"
        >
          <b>
            This is some copy for the body of the Alert
          </b>
        </div>
      </li>
      <li
        onClick={[Function]}
      >
        <svg
          className="close-icon"
          viewBox="0 0 16 16"
        >
          <title>
            
        Close X
    
          </title>
          <path
            className="Close-X_svg__path-01"
            d="M2.948 1.61l5.049 5.058 5.064-5.058a.912.912 0 011.316.027.99.99 0 01-.029 1.37l-4.996 4.99 5.046 5.066a.92.92 0 01-.02 1.32.974.974 0 01-1.357-.02L7.997 9.352 2.956 14.39a.945.945 0 01-1.347-.01.972.972 0 01.013-1.367l5.04-5.012-5.049-5.042a.954.954 0 01-.004-1.346.944.944 0 011.339-.003z"
          />
        </svg>
      </li>
    </ul>
    <div
      className="alert-footer"
    >
      <h6>
        Footer
      </h6>
    </div>
  </div>
</div>
`;

exports[`Alert:.......................SnapShot Testing: Alert with header matches jest snapshot 1`] = `
<div
  className="alert error "
  role="alert"
>
  <div
    className="alert-header error"
  >
    <span
      className="header"
    />
  </div>
  <div
    className="alert-body"
  >
    <ul
      className="layout"
    >
      <li>
        <svg
          className="alert-icon"
          viewBox="0 0 32 32"
        >
          <title>
            
        Error Notification
    
          </title>
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              className="Notification-Error-Light_svg__path-01"
              d="M21.744 29.867l8.123-8.123V10.256l-8.123-8.123H10.256l-8.123 8.123v11.488l8.123 8.123h11.488zM8.956 31.583l-8.331-8.33c-.4-.4-.625-.944-.625-1.51V10.257c0-.565.225-1.108.625-1.508L8.748.625c.4-.4.943-.625 1.508-.625h11.488c.566 0 1.109.225 1.509.625l8.122 8.123c.4.4.625.943.625 1.508v11.488c0 .565-.225 1.108-.625 1.508l-8.122 8.123c-.4.4-.943.625-1.509.625H9.962c-.377 0-.74-.15-1.006-.417z"
            />
            <path
              className="Notification-Error-Light_svg__path-02"
              d="M17.27 20.64a.825.825 0 01-.145.488.52.52 0 01-.171.16.356.356 0 01-.174.045h-1.56c-.133 0-.248-.068-.345-.205a.825.825 0 01-.145-.489l-.508-12.834c0-.188.049-.351.145-.488.097-.137.212-.206.345-.206h2.576c.133 0 .248.069.345.206.096.137.145.3.145.488L17.27 20.64zM16 25.6a1.422 1.422 0 110-2.844 1.422 1.422 0 010 2.844z"
            />
          </g>
        </svg>
      </li>
      <li>
        <div
          className="message"
        >
          <b>
            This is some copy for the body of the Alert
          </b>
        </div>
      </li>
      <li
        onClick={[Function]}
      >
        <svg
          className="close-icon"
          viewBox="0 0 16 16"
        >
          <title>
            
        Close X
    
          </title>
          <path
            className="Close-X_svg__path-01"
            d="M2.948 1.61l5.049 5.058 5.064-5.058a.912.912 0 011.316.027.99.99 0 01-.029 1.37l-4.996 4.99 5.046 5.066a.92.92 0 01-.02 1.32.974.974 0 01-1.357-.02L7.997 9.352 2.956 14.39a.945.945 0 01-1.347-.01.972.972 0 01.013-1.367l5.04-5.012-5.049-5.042a.954.954 0 01-.004-1.346.944.944 0 011.339-.003z"
          />
        </svg>
      </li>
    </ul>
    <div
      className="alert-footer"
    >
      
    </div>
  </div>
</div>
`;

exports[`Alert:.......................SnapShot Testing: Error Alert Matches jest snapshot 1`] = `
<div
  className="alert error "
  role="alert"
>
  <div
    className="alert-header error"
  >
    <span
      className="header"
    />
  </div>
  <div
    className="alert-body"
  >
    <ul
      className="layout"
    >
      <li>
        <svg
          className="alert-icon"
          viewBox="0 0 32 32"
        >
          <title>
            
        Error Notification
    
          </title>
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              className="Notification-Error-Light_svg__path-01"
              d="M21.744 29.867l8.123-8.123V10.256l-8.123-8.123H10.256l-8.123 8.123v11.488l8.123 8.123h11.488zM8.956 31.583l-8.331-8.33c-.4-.4-.625-.944-.625-1.51V10.257c0-.565.225-1.108.625-1.508L8.748.625c.4-.4.943-.625 1.508-.625h11.488c.566 0 1.109.225 1.509.625l8.122 8.123c.4.4.625.943.625 1.508v11.488c0 .565-.225 1.108-.625 1.508l-8.122 8.123c-.4.4-.943.625-1.509.625H9.962c-.377 0-.74-.15-1.006-.417z"
            />
            <path
              className="Notification-Error-Light_svg__path-02"
              d="M17.27 20.64a.825.825 0 01-.145.488.52.52 0 01-.171.16.356.356 0 01-.174.045h-1.56c-.133 0-.248-.068-.345-.205a.825.825 0 01-.145-.489l-.508-12.834c0-.188.049-.351.145-.488.097-.137.212-.206.345-.206h2.576c.133 0 .248.069.345.206.096.137.145.3.145.488L17.27 20.64zM16 25.6a1.422 1.422 0 110-2.844 1.422 1.422 0 010 2.844z"
            />
          </g>
        </svg>
      </li>
      <li>
        <div
          className="message"
        >
          <b>
            This is some copy for the body of the Alert
          </b>
        </div>
      </li>
      <li
        onClick={[Function]}
      >
        <svg
          className="close-icon"
          viewBox="0 0 16 16"
        >
          <title>
            
        Close X
    
          </title>
          <path
            className="Close-X_svg__path-01"
            d="M2.948 1.61l5.049 5.058 5.064-5.058a.912.912 0 011.316.027.99.99 0 01-.029 1.37l-4.996 4.99 5.046 5.066a.92.92 0 01-.02 1.32.974.974 0 01-1.357-.02L7.997 9.352 2.956 14.39a.945.945 0 01-1.347-.01.972.972 0 01.013-1.367l5.04-5.012-5.049-5.042a.954.954 0 01-.004-1.346.944.944 0 011.339-.003z"
          />
        </svg>
      </li>
    </ul>
    <div
      className="alert-footer"
    >
      
    </div>
  </div>
</div>
`;

exports[`Alert:.......................SnapShot Testing: Info Alert Matches jest snapshot 1`] = `
<div
  className="alert info "
  role="alert"
>
  <div
    className="alert-header info"
  >
    <span
      className="header"
    />
  </div>
  <div
    className="alert-body"
  >
    <ul
      className="layout"
    >
      <li>
        <svg
          className="alert-icon"
          viewBox="0 0 32 32"
        >
          <title>
            
        Information Notification
    
          </title>
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              className="Notification-Info-Light_svg__path-01"
              d="M16 29.867c7.658 0 13.867-6.209 13.867-13.867 0-7.658-6.209-13.867-13.867-13.867C8.342 2.133 2.133 8.342 2.133 16c0 7.658 6.209 13.867 13.867 13.867zM16 32C7.163 32 0 24.837 0 16S7.163 0 16 0s16 7.163 16 16-7.163 16-16 16z"
            />
            <path
              className="Notification-Info-Light_svg__path-02"
              d="M16 11.022a1.6 1.6 0 100-3.2 1.6 1.6 0 000 3.2zm1.422 13.895a.71.71 0 01-.175.48.533.533 0 01-.416.203H15.17a.533.533 0 01-.416-.202.71.71 0 01-.175-.481v-11.79a.71.71 0 01.175-.48.533.533 0 01.416-.203h1.662c.16 0 .3.068.416.203a.71.71 0 01.175.48v11.79z"
            />
          </g>
        </svg>
      </li>
      <li>
        <div
          className="message"
        >
          <b>
            This is some copy for the body of the Alert
          </b>
        </div>
      </li>
      <li
        onClick={[Function]}
      >
        <svg
          className="close-icon"
          viewBox="0 0 16 16"
        >
          <title>
            
        Close X
    
          </title>
          <path
            className="Close-X_svg__path-01"
            d="M2.948 1.61l5.049 5.058 5.064-5.058a.912.912 0 011.316.027.99.99 0 01-.029 1.37l-4.996 4.99 5.046 5.066a.92.92 0 01-.02 1.32.974.974 0 01-1.357-.02L7.997 9.352 2.956 14.39a.945.945 0 01-1.347-.01.972.972 0 01.013-1.367l5.04-5.012-5.049-5.042a.954.954 0 01-.004-1.346.944.944 0 011.339-.003z"
          />
        </svg>
      </li>
    </ul>
    <div
      className="alert-footer"
    >
      
    </div>
  </div>
</div>
`;

exports[`Alert:.......................SnapShot Testing: Success Alert Matches jest snapshot 1`] = `
<div
  className="alert success "
  role="alert"
>
  <div
    className="alert-header success"
  >
    <span
      className="header"
    />
  </div>
  <div
    className="alert-body"
  >
    <ul
      className="layout"
    >
      <li>
        <svg
          className="alert-icon"
          viewBox="0 0 32 32"
        >
          <title>
            
        Success Notification
    
          </title>
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              className="Notification-Success-Light_svg__path-01"
              d="M16 29.867c7.658 0 13.867-6.209 13.867-13.867 0-7.658-6.209-13.867-13.867-13.867C8.342 2.133 2.133 8.342 2.133 16c0 7.658 6.209 13.867 13.867 13.867zM16 32C7.163 32 0 24.837 0 16S7.163 0 16 0s16 7.163 16 16-7.163 16-16 16z"
            />
            <path
              className="Notification-Success-Light_svg__path-02"
              d="M12.936 19.61l9.71-9.984a1.422 1.422 0 112.039 1.983L12.993 23.63l-6.1-5.927a1.422 1.422 0 011.981-2.04l4.062 3.946z"
            />
          </g>
        </svg>
      </li>
      <li>
        <div
          className="message"
        >
          <b>
            This is some copy for the body of the Alert
          </b>
        </div>
      </li>
      <li
        onClick={[Function]}
      >
        <svg
          className="close-icon"
          viewBox="0 0 16 16"
        >
          <title>
            
        Close X
    
          </title>
          <path
            className="Close-X_svg__path-01"
            d="M2.948 1.61l5.049 5.058 5.064-5.058a.912.912 0 011.316.027.99.99 0 01-.029 1.37l-4.996 4.99 5.046 5.066a.92.92 0 01-.02 1.32.974.974 0 01-1.357-.02L7.997 9.352 2.956 14.39a.945.945 0 01-1.347-.01.972.972 0 01.013-1.367l5.04-5.012-5.049-5.042a.954.954 0 01-.004-1.346.944.944 0 011.339-.003z"
          />
        </svg>
      </li>
    </ul>
    <div
      className="alert-footer"
    >
      
    </div>
  </div>
</div>
`;

exports[`Alert:.......................SnapShot Testing: Warning Alert Matches jest snapshot 1`] = `
<div
  className="alert warning "
  role="alert"
>
  <div
    className="alert-header warning"
  >
    <span
      className="header"
    />
  </div>
  <div
    className="alert-body"
  >
    <ul
      className="layout"
    >
      <li>
        <svg
          className="alert-icon"
          viewBox="0 0 32 32"
        >
          <title>
            
        Warning Notification
    
          </title>
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              className="Notification-Warning-Light_svg__path-01"
              d="M16.968 2.767a1.084 1.084 0 00-1.938 0L2.282 28.264a1.084 1.084 0 00.97 1.569h25.496a1.084 1.084 0 00.97-1.569L16.967 2.767zm1.939-.97l12.749 25.498A3.251 3.251 0 0128.748 32H3.251a3.251 3.251 0 01-2.908-4.705L13.091 1.798a3.251 3.251 0 015.816 0z"
            />
            <path
              className="Notification-Warning-Light_svg__path-02"
              d="M17.27 22.417a.825.825 0 01-.145.488.52.52 0 01-.171.162.356.356 0 01-.174.044h-1.56c-.133 0-.248-.068-.345-.206a.825.825 0 01-.145-.488l-.508-12.123c0-.189.049-.351.145-.488.097-.137.212-.206.345-.206h2.576c.133 0 .248.069.345.206.096.137.145.3.145.488l-.508 12.123zM16 27.377a1.422 1.422 0 110-2.844 1.422 1.422 0 010 2.845z"
            />
          </g>
        </svg>
      </li>
      <li>
        <div
          className="message"
        >
          <b>
            This is some copy for the body of the Alert
          </b>
        </div>
      </li>
      <li
        onClick={[Function]}
      >
        <svg
          className="close-icon"
          viewBox="0 0 16 16"
        >
          <title>
            
        Close X
    
          </title>
          <path
            className="Close-X_svg__path-01"
            d="M2.948 1.61l5.049 5.058 5.064-5.058a.912.912 0 011.316.027.99.99 0 01-.029 1.37l-4.996 4.99 5.046 5.066a.92.92 0 01-.02 1.32.974.974 0 01-1.357-.02L7.997 9.352 2.956 14.39a.945.945 0 01-1.347-.01.972.972 0 01.013-1.367l5.04-5.012-5.049-5.042a.954.954 0 01-.004-1.346.944.944 0 011.339-.003z"
          />
        </svg>
      </li>
    </ul>
    <div
      className="alert-footer"
    >
      
    </div>
  </div>
</div>
`;
